lib/shrinkwrap.js

```diff
@@ -1,5 +1,7 @@
 const { resolve, basename } = require('path')
-const { promises: { unlink } } = require('fs')
+const util = require('util')
+const fs = require('fs')
+const { unlink } = fs.promises || util.promisify(fs.unlink)
```

Comment:
How does one destructure an `unlink` property from a promisified function value? It's not following the pattern of [lib/utils/reify-finish.js](https://github.com/npm/cli/pull/2654/files#diff-21fa145c2edaa021d8e75bc1410627286523039917e914bb78604be144a70523) (or any number of dependency updates to address missing `fs.promises`), so maybe the tests are secretly failing for shrinkwrap? "Not throwing" isn't the same as "correct behavior in alternate circumstance".

```patch
-const { unlink } = fs.promises || util.promisify(fs.unlink)
+const { unlink } = fs.promises || { unlink: util.promisify(fs.unlink) }
```

Reply:
whoops, you're right. this was a mistake. i'll put up a quick PR to fix it.

Reply:
Note that tests aren't actually running in node 10.0, so the failures there wouldn't have shown up in CI.

Reply:
https://github.com/npm/cli/pull/2688